src/OrchardCore.Modules/OrchardCore.Users/Services/ExternalLoginFormEvents.cs

```diff
@@ -36,10 +42,30 @@ public override async Task<IActionResult> LoggingInAsync()
             return null;
         }
 
+        // To prevent infinite redirects, we store an ELAR cookie, which stands for ExternalLoginAutoRedirect.
+        // This cookie helps us avoid executing the same redirect multiple times. 
+        // The acronym ELAR has no special meaning, and any other value could be used in its place; it is chosen simply for brevity.
+        if (_httpContextAccessor.HttpContext.Request.Cookies.TryGetValue(ExternalLoginAutoRedirectCookieName, out var _))
+        {
+            _httpContextAccessor.HttpContext.Response.Cookies.Delete(ExternalLoginAutoRedirectCookieName);
+
+            return null;
+        }
```

Comment:
When can such an infinite redirect happen, why is this cookie necessary? I tried external login with Entra ID before this change and it didn't cause a redirect loop (though `UseExternalProviderIfOnlyOneDefined` was on the default `false`, i.e. this method short-circuited).

Also, instead of cookies, you can add a query string parameter, which might be safer than an external login (cookies might not be accepted), and we're not saving something in a cookie that needn't be persistent beyond the next request.

Additionally, if one signs out and then in again in the same browser (or just gets signed out when closing it, given that this cookie will persist beyond browser close but the user session doesn't) the cookie will remain there, and the logic form line 69 won't run, but I think it should. I think this will be the case too if a login provider is added and the user wants to use it but won't be able to.

Reply:
The issue is explained in #17059. I can change this into using query params.

Reply:
I see, OK then. Yes, I'd prefer a query string parameter (or perhaps even `TempData`, but I'm wary of that) over cookies.

Reply:
@Piedone query-parameter did not work, It gets lost in the redirects. I did use TempData instead. Let me know if you see anything else or if you approve this.